manifests/init.pp

```diff
@@ -438,6 +480,123 @@
       enable  => $enable_pcsd_service,
       require => Package['pcs'],
     }
+
+    # Determine if this node should perform authorizations
+    case $manage_pcsd_auth_node {
+      'first': { $auth_node = $quorum_members[0] }
+      'last': { $auth_node = $quorum_members[length($quorum_members)-1] }
+      default: {}
+    }
+
+    # If the local data matches auth_node (hostname or primary IP) we can
+    # perform auth processing for subsequent components
+    if $trusted['certname'] == $auth_node or
+        $facts['networking']['ip'] == $auth_node {
```

Comment:
I'm not sure I follow your suggestion here. The purpose of this check is to determine whether the current node should attempt to perform the authorizations. The list of quorum members has the same ordering whether it is comprised of IPs or host names or a mixture of both. Are you suggesting checking each IP the current node has? 

Originally I limited it to the default IP for simplicity, however, it could definitely iterate through all IPs via something like this:
```puppet
$local_ip_addresses = $facts['networking']['interfaces'].map | $entry | {
  if 'ip' in $entry[1] { 
    $entry[1]['ip']
  }
}

if $trusted['certname'] == $auth_node or
  $auth_node in $local_ip_addresses {
```

Reply:
Yes, sorry; I've had some setups where the 'primary' default route is some WAN address, but the pacemaker cluster was running totally on other interfaces with a LAN address. And maybe the iteration you suggest can be ignored by using something like `stdlib::grep`? (Or `filter`, if we're assuming Puppet ≥ 4.0)

Reply:
Take a look at my latest commit - I think what I've implemented here should work for the cases we've been discussing.